browser-switch/src/main/java/com/braintreepayments/api/BrowserSwitchClient.java

```diff
@@ -168,4 +163,47 @@ BrowserSwitchResult getResult(@NonNull FragmentActivity activity) {
 
         return result;
     }
+
+    /**
+     * Deliver a pending browser switch result that was previously captured by another Android activity.
+     * <p>
+     * Success results will be delivered only once. If there are no pending
+     * browser switch results in the cache, this method does nothing.
+     *
+     * @param context the context used to access the cache
+     * @return {@link BrowserSwitchResult}
+     */
+    public BrowserSwitchResult deliverResultFromCache(@NonNull Context context) {
+        Context applicationContext = context.getApplicationContext();
+        BrowserSwitchResult result = persistentStore.getActiveResult(applicationContext);
+        if (result != null) {
+            persistentStore.removeAll(applicationContext);
+        }
+        return result;
+    }
+
+    /**
+     * Capture a pending browser switch result for an Android activity.
+     * <p>
+     * To obtain the result in a separate activity, call {@link #deliverResultFromCache(Context)}.
+     *
+     * @param activity the activity that received the deep link back into the app
+     */
+    public void captureResult(@NonNull FragmentActivity activity) {
```

Comment:
Is `capture` a common term when using Android's `SharedPreferences`? If not, I think `persist` or `store` more clearly communicate the action here.

Reply:
Yeah `put` would be the most direct alignment in terms of verbs.

Here's the invocation in the context of the [Core SDK PR](https://github.com/braintree/braintree_android/blob/269f207a4e603328a526ec75173715001bb59ec5/BraintreeCore/src/main/java/com/braintreepayments/api/BraintreeDeepLinkActivity.java#L15). I'd be down for `persist` or `store`. Capture comes from it being called in an invisible activity, but the naming doesn't have to align with Core.

Reply:
Gotcha. I don't feel too strongly, but wanted to ask. 

If we stick with `capture`, could we clarify in the docstring `* Capture a pending browser switch result for an Android activity into persistent storage.`?

Reply:
Yeah definitely. That's a good call I'll update the docs string.